src/main/java/com/firebolt/jdbc/statement/preparedstatement/FireboltBackendPreparedStatement.java

```
@@ -0,0 +1,173 @@
+            //this is used as a supported target type verification. null is not supported
+            if (x != null) {
+                JavaTypeToFireboltSQLString.getType(targetSqlType, x);
+            }
+            providedParameters.put(parameterIndex, x);
+        } catch (FireboltException fbe) {
+            if (ExceptionType.TYPE_NOT_SUPPORTED.equals(fbe.getType())) {
+                throw new SQLFeatureNotSupportedException(fbe.getMessage(), fbe);
+            }
+            throw fbe;
+        }
+    }
+
+    @Override
+    public void setObject(int parameterIndex, Object x) throws SQLException {
+        validateStatementIsNotClosed();
+        JavaTypeToFireboltSQLString.validateObjectIsOfSupportedType(x);
+        providedParameters.put(parameterIndex, x);
+    }
+
+    @Override
+    public void setNull(int parameterIndex, int sqlType, String typeName) throws SQLException {
+        validateStatementIsNotClosed();
+        providedParameters.put(parameterIndex, null);
+    }
+
+    @Override
+    public void setArray(int parameterIndex, Array x) throws SQLException {
+        validateStatementIsNotClosed();
+        providedParameters.put(parameterIndex, new JSONArray(x.getArray()).toString());
+    }
+
+    @Override
+    public void setObject(int parameterIndex, Object x, int targetSqlType, int scaleOrLength) throws SQLException {
+        validateStatementIsNotClosed();
+        try {
+            boolean isNumber = (DECIMAL == targetSqlType || NUMERIC == targetSqlType) && x instanceof Number;
+            //convert object to a string if it is not a number
+            String str = isNumber ? formatDecimalNumber(x, scaleOrLength) : JavaTypeToFireboltSQLString.transformAny(x, targetSqlType);
+            providedParameters.put(parameterIndex, Double.valueOf(str));
+        } catch (FireboltException fbe) {
+            if (ExceptionType.TYPE_NOT_SUPPORTED.equals(fbe.getType())) {
```

if there is any other type of exception seems like we are swallowing it. Let's throw a sqlexception and add the message inside.